### DISCUSSION TITLE
Getting Class 'DOMDocument' not found exception

### DISCUSSION THREAD

**Discussion by mdcclxv** [YOURLS/YOURLS#3069] [Q&A] [answered]
When performing the following curl:

```
curl -X POST -H "Content-Type: application/json" -d @url.json https://site.mobi/yourls-api.php
```
with the following url.json file:
```
{
"signature": "1212121212",
"action": "shorturl",
"format": "json",
"url" : "https://anothersite.mobi/?page=external_user_review"
}
```
I get this in apache's error.log:
```
PHP Fatal error:  Uncaught Error: Class 'DOMDocument' not found in /home/ubuntu/app/includes/vendor/spatie/array-to-xml/src/ArrayToXml.php:29
Stack trace:
#0 /home/ubuntu/app/includes/vendor/spatie/array-to-xml/src/ArrayToXml.php(67): Spatie\\ArrayToXml\\ArrayToXml->__construct()
#1 /home/ubuntu/app/includes/functions.php(77): Spatie\\ArrayToXml\\ArrayToXml::convert()
#2 /home/ubuntu/app/includes/functions-api.php(142): yourls_xml_encode()
#3 /home/ubuntu/app/includes/auth.php(17): yourls_api_output()
#4 /home/ubuntu/app/includes/functions-auth.php(14): require_once('/home/ubunt...')
#5 /home/ubuntu/app/yourls-api.php(12): yourls_maybe_require_auth()
#6 {main}
thrown in /home/ubuntu/app/includes/vendor/spatie/array-to-xml/src/ArrayToXml.php on line 29
```

Any idea what's the catch?

**Comment by mdcclxv** [comment, marked as answer]
I've solved it in the meantime, it was php-xml module that was missing. Installed that with ```sudo apt install php-xml``` and everything works fine now. For anyone having the same issue.

**Reply by dgw** [reply]
Glad to hear the issue is resolved!

@ozh I can't find a list of required/expected PHP modules. Maybe that needs to be added [here in the wiki](https://github.com/YOURLS/YOURLS/wiki/Requirements)?

**Reply by ozh** [reply]
Well, php-xml is [enabled by default](https://www.php.net/manual/en/xml.installation.php), so I'm not sure it's practical to list all extensions that "should be there unless the user explicitly omits them"

**Comment by mdcclxv** [comment]
I don't think that "enabled by default" is always true. Nobody disabled that extension on this server, that I can tell for sure. This is an Ubuntu server instance on Google's cloud, formerly 16, then upgraded to 18 and then to 20. So, maybe this extension being disabled is a legacy? Or an intended omission on Google's side?

Anyway, a detailed list of requirements is always welcome, can do no harm.
